### ISSUE TITLE
Potential problem with distance.IdentityFakeDistance

### ISSUE THREAD

**Issue by frank-pieterse** [ICB-DCM/pyABC#380]
Hi, sorry if this is not the way to report a bug, I'm new to github.

I'm using pyabc for a project in university, and i ran into some errors. My model function outputs a distance, so i used the IdentityFakeDistance method. In the API, the return type is mentioned:

Return type

    float

However, when i looked inside the code, the class looks like this:
```
class IdentityFakeDistance(Distance):
    """
    A fake distance function, which just passes the summary statistics on.
    This class assumes that the model already returns the distance. This can be
    useful in cases where simulating can be stopped early, when during the
    simulation some condition is reached which makes it impossible to accept
    the particle.
    """

    def __call__(self,
                 x: dict,
                 x_0: dict,
                 t: int = None,
                 par: dict = None) -> float:
         return x
```
x is a dictionary, and it returns x. So the return type is dict. This cause an error down the line, when the algorithm was sorting the weighted distances, which it exptected to be of type float. I changed your code to:

```
class IdentityFakeDistance(Distance):
    """
    A fake distance function, which just passes the summary statistics on.
    This class assumes that the model already returns the distance. This can be
    useful in cases where simulating can be stopped early, when during the
    simulation some condition is reached which makes it impossible to accept
    the particle.
    """

    def __call__(self,
                 x: dict,
                 x_0: dict,
                 t: int = None,
                 par: dict = None) -> float:
        return list(x.values())[0]
```
After my change, the algorithm seems to run as intended. I think this was a bug, since all other distance functions return floats.

**Comment by yannikschaelte** [comment]
Hi @frank-pieterse , so your model already computes the distances and returns a dict (which can be stored) with one entry whose value is the distance? In that case I think that defining the distance exactly the way you did is accurate.

The IdentityFakeDistance was primarily intended to be used together with an [IntegratedModel](https://pyabc.readthedocs.io/en/latest/api_model.html#pyabc.model.IntegratedModel. In that case, it is however never directly invoked. Therefore I think the correct behavior for that class would be to have a distance that raises if invoked, or alternatively do what you proposed with some checks. Thanks for pointing this out!

**Comment by frank-pieterse** [comment]
Hi @yannikschaelte, thanks for the quick response! My model indeed returns a dict with just the distance in it.

**Comment by yannikschaelte** [comment]
Hi @frank-pieterse , sorry I lost track over the holidays. Is this problem resolved for you if you use a customized distance?

**Comment by yannikschaelte** [comment]
fixed in #388
